# Try Again

If you take a look at various commands’ description, you will notice that a vast majority of them share the [same arguments](appendices/common-arguments.md). Four of these arguments are targeted at error handling: `errorcall`, `errorjump`, `errormessage` and `errorresult`. You can use one of them to convince a user to provide a correct input.

You will use the `try` command, which is another block command: it tries to execute everything inside the block between `try` and `end` keywords. In case an error occurs within this block, the robot will do what one of the `error...` arguments instructs: call a procedure, jump to a label, display a message or just store the error information in a variable.

Try this out:

```G1ANT
try errorcall askUser
  call askUser
  ♥input = ⟦integer⟧♥input
end

procedure askUser
  dialog.ask ‴Enter a digit (0-9):‴ result ♥input
end
```

The `try`…`end` block calls the `askUser` procedure, where a user is asked to enter a digit in a dialog box. This digit is stored in the `♥input` variable.

When the `askUser` procedure ends, the robot goes back to the place where the procedure was called, then it tries to force the integer type on the provided value. If it fails to do so, because a user entered non-numerical characters, an error occurs. And this is the moment, when the `errorcall` argument for the `try` command steps in: it tells the robot to execute the `askUser` procedure again in hope for a correct user input. This will be repeated until a user provides digits and the `♥input` variable is successfully converted into an integer.

In the next lesson you will see how you can 